lib/network/modules/InteractionHandler.js

```diff
@@ -205,21 +205,19 @@ class InteractionHandler {
       selected = true;
     }
 
-    if (selectedEdgesCount - previouslySelectedEdgeCount > 0) { // edge was selected
-      this.selectionHandler._generateClickEvent('selectEdge', event, pointer);
-      selected = true;
-    }
-    else if (selectedEdgesCount - previouslySelectedEdgeCount < 0) { // edge was deselected
-      this.selectionHandler._generateClickEvent('deselectEdge', event, pointer, previousSelection);
-      selected = true;
+    if (selectedEdgesCount - previouslySelectedEdgeCount < 0 || (selectedEdgesCount === previouslySelectedEdgeCount && edgesChanges === true)) { // edge was deselected
+        this.selectionHandler._generateClickEvent('deselectEdge', event, pointer, previousSelection);
+        selected = true;
     }
-    else if (selectedEdgesCount === previouslySelectedEdgeCount && edgesChanges === true) {
-      this.selectionHandler._generateClickEvent('deselectEdge', event, pointer, previousSelection);
+
+    // Check if the user clicked on an edge
+    let selectedNode = this.selectionHandler._getSelectedNode();
+    let selectedEdge = this.selectionHandler._getSelectedEdge();
+    if (selectedNode === undefined && selectedEdge !== undefined) {
```

Comment:
I don't think you have covered the case where you have 4 edges selected (multiselect), then deselect one. What happens now is that you fire a deselect AND a select edge event. You should add the node check in the if statement above.